src/OpenTelemetry.Api/Context/Propagation/CompositeTextMapPropagator.cs

```diff
@@ -68,9 +68,9 @@ public CompositeTextMapPropagator(IEnumerable<TextMapPropagator> propagators)
     /// <inheritdoc/>
     public override PropagationContext Extract<T>(PropagationContext context, T carrier, Func<T, string, IEnumerable<string>?> getter)
     {
-        for (int i = 0; i < this.propagators.Count; i++)
+        foreach (var propagator in this.propagators)
```

Comment:
The problem here is when you `foreach` it is a call to `_thing_.GetEnumerator()`. If `thing` is an interface like `IEnumerable` or `IReadOnlyList` then `GetEnumerator()` usually returns an `IEnumerator`. The implementation for the enumerator on the concrete type is usually a `struct` so converting it to `IEnumerator` causes a boxing allocation. Agree the `foreach` is more readable, but the `for`s are written specifically to cheat and avoid these allocations. Avoiding allocations is a great good than readability in this case IMO.

Reply:
fair enough. will close